source/CompactRegular.lagda

```diff
@@ -0,0 +1,402 @@
+good idea to address this duplication at some point.
+
+\begin{code}
+
+is-directed : (P : poset 𝓤 𝓥) → (S : Fam 𝓦 ∣ P ∣ₚ) → Ω (𝓥 ⊔ 𝓦)
+is-directed P (I , s) =
+   ∥ I ∥Ω
+ ∧ (Ɐ i ∶ I , Ɐ j ∶ I , Ǝ k ∶ I , ((s i ≤ s k) ∧ (s j ≤ s k)) holds)
+  where open PosetNotation P using (_≤_)
+
+\end{code}
+
+\begin{code}
+
+way-below : (F : frame 𝓤 𝓥 𝓦) → ⟨ F ⟩ → ⟨ F ⟩ → Ω (𝓤 ⊔ 𝓥 ⊔ 𝓦 ⁺)
+way-below {𝓤 = 𝓤} {𝓦 = 𝓦} F U V =
+ Ɐ S ∶ Fam 𝓦 ⟨ F ⟩ , is-directed (poset-of F) S ⇒
+  V ≤ (⋁[ F ] S) ⇒ (Ǝ i ∶ index S , (U ≤ S [ i ]) holds)
+   where
+    open PosetNotation (poset-of F) using (_≤_)
+
+infix 5 way-below
+
+syntax way-below F U V = U ≪[ F ] V
+
+\end{code}
+
+\begin{code}
+
+is-compact-open : (F : frame 𝓤 𝓥 𝓦) → ⟨ F ⟩ → Ω (𝓤 ⊔ 𝓥 ⊔ 𝓦 ⁺)
+is-compact-open F U = U ≪[ F ] U
+
+\end{code}
+
+A compact frame is simply a frame whose top element is finite.
+
+\begin{code}
+
+is-compact : frame 𝓤 𝓥 𝓦 → Ω (𝓤 ⊔ 𝓥 ⊔ 𝓦 ⁺)
+is-compact F = is-compact-open F 𝟏[ F ]
+
+\end{code}
+
+\begin{code}
+
+compacts-are-closed-under-joins : (F : frame 𝓤 𝓥 𝓦)
+                                → (U V : ⟨ F ⟩)
+                                → is-compact-open F U holds
+                                → is-compact-open F V holds
+                                → is-compact-open F (U ∨[ F ] V) holds
+compacts-are-closed-under-joins F U V κ₁ κ₂ S dir@(_ , up) p =
+ ∥∥-rec₂ ∃-is-prop γ s₁′ s₂′
+  where
+   open PosetNotation  (poset-of F) using (_≤_)
+   open PosetReasoning (poset-of F)
+
+   -- S covers U.
+   q₁ : (U ≤ (⋁[ F ] S)) holds
+   q₁ = U ≤⟨ ∨[ F ]-upper₁ U V ⟩ U ∨[ F ] V ≤⟨ p ⟩ ⋁[ F ] S ■
+
+   -- S covers V.
+   q₂ : (V ≤ (⋁[ F ] S)) holds
+   q₂ = V ≤⟨ ∨[ F ]-upper₂ U V ⟩ U ∨[ F ] V ≤⟨ p ⟩ ⋁[ F ] S ■
+
+   s₁′ : ∥ Σ i₁ ꞉ index S , (U ≤ S [ i₁ ]) holds ∥
+   s₁′ = κ₁ S dir q₁
+
+   s₂′ : ∥ Σ i₂ ꞉ index S , (V ≤ S [ i₂ ]) holds ∥
+   s₂′ = κ₂ S dir q₂
+
+   γ : (Σ i₁ ꞉ index S , (U ≤ S [ i₁ ]) holds)
+     → (Σ i₂ ꞉ index S , (V ≤ S [ i₂ ]) holds)
+     → ∃ i ꞉ index S , ((U ∨[ F ] V) ≤ S [ i ]) holds
+   γ (i₁ , s₁) (i₂ , s₂) = ∥∥-rec ∃-is-prop δ (up i₁ i₂)
+    where
+     δ : Σ i ꞉ index S , ((S [ i₁ ] ≤ S [ i ]) ∧ (S [ i₂ ] ≤ S [ i ])) holds
+       → ∃ i ꞉ index S , ((U ∨[ F ] V) ≤ S [ i ]) holds
+     δ (i , r₁ , r₂) = ∣ i , ∨[ F ]-least ε ζ ∣
+      where
+       ε : (U ≤ S [ i ]) holds
+       ε = U ≤⟨ s₁ ⟩ S [ i₁ ] ≤⟨ r₁ ⟩ S [ i ] ■
+
+       ζ : (V ≤ S [ i ]) holds
+       ζ = V ≤⟨ s₂ ⟩ S [ i₂ ] ≤⟨ r₂ ⟩ S [ i ] ■
+
+\end{code}
+
+\section{Well Inside}
+
+\begin{code}
+
+-- The type family expressing that a given open of a frame is clopen.
+well-inside₀ : (F : frame 𝓤 𝓥 𝓦) → ⟨ F ⟩ → ⟨ F ⟩ → 𝓤 ̇ 
+well-inside₀ F U V =
+ Σ W ꞉ ⟨ F ⟩ , (U ∧[ F ] W ≡ 𝟎[ F ]) × (V ∨[ F ] W ≡ 𝟏[ F ])
+
+infix 4 well-inside₀
+
+syntax well-inside₀ F U V = U ⋜₀[ F ] V
+
+-- The well inside relation is not propositional in general, even though the
+-- “has complement” (i.e. is well inside itself) is propositional.
+well-inside-is-not-prop : is-univalent 𝓤₀
```

Comment:
well-inside in the name should have subscript 0.

Reply:
Done ✅